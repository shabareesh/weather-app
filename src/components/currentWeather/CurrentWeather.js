import React from 'react';
import PropTypes from 'prop-types';

import {formatCurrentWeatherDate} from "../../utils/utils";
import './currentWeather.scss';

const CurrentWeather = ({ weatherInfo:{city_name, state_code, country_code, data} }) => (
  <div className="current-weather">
    <h4 className="current-weather--city">{city_name} - {state_code}, {country_code}</h4>
    <p className="current-weather--date">{formatCurrentWeatherDate(data[0].datetime)}</p>
    <p className="current-weather--highTemp"><strong>H {data[0].high_temp}<sup>o</sup></strong><span>/ L {data[0].low_temp}<sup>o</sup></span></p>
    <p className="current-weather--description">{data[0].weather.description}</p>
  </div>
);

CurrentWeather.propTypes = {
  weatherInfo: PropTypes.shape({
    city_name:PropTypes.string.isRequired,
    state_code:PropTypes.string.isRequired,
    country_code:PropTypes.string.isRequired,
    data:PropTypes.arrayOf(PropTypes.shape({
      high_temp:PropTypes.number.isRequired,
      low_temp:PropTypes.number.isRequired,
      weather:PropTypes.shape({
        description:PropTypes.string.isRequired,
      }).isRequired,
    })),
  }).isRequired,
};

export default CurrentWeather;
