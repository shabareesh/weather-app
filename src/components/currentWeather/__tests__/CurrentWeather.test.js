import React from 'react';
import { render } from "@testing-library/react";

import CurrentWeather from "../CurrentWeather";

describe('CurrentWeather', () => {
  const props = {
    weatherInfo:{
      city_name:'Chicago',
      state_code:'1234',
      country_code: 'US',
      data: [
        {
          high_temp: 45.1,
          low_temp: 45.1,
          weather: {
            description:'some desc'
          },
        },
      ],
    },
  };

  test('should match snapshot', () => {
    const { baseElement } = render(<CurrentWeather {...props} />);

    expect(baseElement).toMatchSnapshot();
  });
});
