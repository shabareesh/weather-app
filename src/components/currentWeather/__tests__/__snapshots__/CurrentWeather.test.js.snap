// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrentWeather should match snapshot 1`] = `
<body>
  <div>
    <div
      class="current-weather"
    >
      <h4
        class="current-weather--city"
      >
        Chicago
         - 
        1234
        , 
        US
      </h4>
      <p
        class="current-weather--date"
      >
        undefined, undefined NaN, NaN
      </p>
      <p
        class="current-weather--highTemp"
      >
        <strong>
          H 
          45.1
          <sup>
            o
          </sup>
        </strong>
        <span>
          / L 
          45.1
          <sup>
            o
          </sup>
        </span>
      </p>
      <p
        class="current-weather--description"
      >
        some desc
      </p>
    </div>
  </div>
</body>
`;
